File: lib/Ntreev.Windows.Forms.Grid/GroupRow.cpp
#include "StdAfx.h"
#include "GroupRow.h"

namespace Ntreev { namespace Windows { namespace Forms { namespace Grid
{
    GroupRow::GroupRow(Ntreev::Windows::Forms::Grid::GridControl^ gridControl, GrGroupRow* pGroupRow)
        : RowBase(gridControl, pGroupRow), m_pGroupRow(pGroupRow)
    {

    }
} /*namespace Grid*/ } /*namespace Forms*/ } /*namespace Windows*/ } /*namespace Ntreev*/

File: lib/WinformGridHost/GroupRow.cs
using Ntreev.Library.Grid;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Ntreev.Windows.Forms.Grid
{
    public class GroupRow : RowBase
    {
        private readonly GrGroupRow m_pGroupRow;

        internal GroupRow(GrGroupRow pGroupRow)
            : base(pGroupRow)
        {
            m_pGroupRow = pGroupRow;
        }

        public override string ToString()
        {
            return this.Text;
        }

        public string Text
        {
            get
            {
                GrGroupHeader pGroupHeader = m_pGroupRow.GetLabel();
                return pGroupHeader.Text;
            }
        }
    }
}
